multirpc/go-server/cmd/main.go

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package main
+
+import (
+	"context"
+	"dubbo.apache.org/dubbo-go/v3"
+	_ "dubbo.apache.org/dubbo-go/v3/imports"
+	"dubbo.apache.org/dubbo-go/v3/protocol"
+	"dubbo.apache.org/dubbo-go/v3/registry"
+	"dubbo.apache.org/dubbo-go/v3/server"
+	greet "github.com/apache/dubbo-go-samples/multirpc/proto"
+	"github.com/dubbogo/gost/log/logger"
+)
+
+type GreetMultiRPCServer struct {
+}
+
+func (srv *GreetMultiRPCServer) Greet(ctx context.Context, req *greet.GreetRequest) (*greet.GreetResponse, error) {
+	resp := &greet.GreetResponse{Greeting: req.Name}
+	return resp, nil
+}
+
+type GreetProvider struct {
+}
+
+func (*GreetProvider) SayHello(req string, req1 string, req2 string) (string, error) {
+	return req + req1 + req2, nil
+}
+
+func main() {
+	//triple
+	ins, err := dubbo.NewInstance(
+		dubbo.WithName("dubbo_multirpc_triple_server"),
+		dubbo.WithRegistry(
+			registry.WithZookeeper(),
+			registry.WithAddress("127.0.0.1:2181"),
+		),
+		//dubbo.WithProtocol(protocol.WithDubbo()),
+		dubbo.WithProtocol(protocol.WithTriple()),
+		//dubbo.WithProtocol(protocol.WithJSONRPC()),
+	)
+	if err != nil {
+		panic(err)
+	}
+	//var protocols []string
+	//protocols = append(protocols, "tri", "dubbo", "jsonrpc")
+	srv, err := ins.NewServer(
+		server.WithServerProtocol(
```

Comment:
As we have configured protocols in ```dubbo.Instance``` layer. We do not need to configure protocols here.